sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -151,7 +152,7 @@ public EncryptedBlobAsyncClient buildEncryptedBlobAsyncClient() {
 
         return new EncryptedBlobAsyncClient(getHttpPipeline(),
             String.format("%s/%s/%s", endpoint, containerName, blobName), serviceVersion, accountName, containerName,
-            blobName, snapshot, customerProvidedKey, keyWrapper, keyWrapAlgorithm);
+            blobName, snapshot, customerProvidedKey, keyWrapper, keyWrapAlgorithm, versionId);
```

Comment:
Is it legal for a url to have a snapshot and a versionId? I would guess not, and it may be worth validating that here.

Reply:
Looks like you covered that below :)

Reply:
Correct, that's covered.